builder/Dockerfile.fedora-37

```diff
@@ -0,0 +1,86 @@
+FROM fedora:37
+
+ENV OS_IDENTIFIER fedora-37
+
+RUN dnf -y upgrade \
+    && dnf -y install dnf-plugins-core \
+    && dnf -y install \
+    autoconf \
+    automake \
+    bzip2-devel \
+    cairo-devel \
+    flexiblas-devel \
+    gcc-c++ \
+    gcc-gfortran \
+    java-11-openjdk-devel \
+    libICE-devel \
+    libSM-devel \
+    libX11-devel \
+    libXmu-devel \
+    libXt-devel \
+    libcurl-devel \
+    libicu-devel \
+    libjpeg-devel \
+    libpng-devel \
+    libtiff-devel \
+    libtool \
+    make \
+    ncurses-devel \
+    pango-devel \
+    pcre-devel \
+    pcre2-devel \
+    readline-devel \
+    rpm-build \
+    tcl-devel \
+    tex \
+    texinfo-tex \
+    texlive-collection-latexrecommended \
+    tk-devel \
+    unzip \
+    valgrind-devel \
+    which \
+    wget \
+    xz-devel \
+    zlib-devel \
+    && dnf clean all
+
+# Install AWS CLI
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-$(uname -m).zip" -o "awscliv2.zip" && \
+    unzip awscliv2.zip && \
+    ./aws/install && \
+    rm -rf aws awscliv2.zip
+
+RUN if [ "$(arch)" == "aarch64" ]; then echo arm64; else echo amd64; fi > /tmp/arch
+
+RUN curl -LO "https://github.com/goreleaser/nfpm/releases/download/v2.18.1/nfpm_$(cat /tmp/arch).rpm" && \
+    dnf install -y "nfpm_$(cat /tmp/arch).rpm" && \
+    rm "nfpm_$(cat /tmp/arch).rpm"
+
+RUN chmod 0777 /opt
+
+# Configure flags for RHEL 9 that don't use the defaults in build.sh
+ENV CONFIGURE_OPTIONS="\
+    --enable-R-shlib \
+    --with-tcltk \
+    --enable-memory-profiling \
+    --with-x \
+    --with-system-valgrind-headers \
+    --with-tcl-config=/usr/lib64/tclConfig.sh \
+    --with-tk-config=/usr/lib64/tkConfig.sh \
+    --enable-prebuilt-html"
```

Comment:
For RHEL 9, we had a complicated config split in R 4.3.0 to switch to FlexiBLAS instead of OpenBLAS, because R <= 4.2 were built with OpenBLAS and we didn't want to break users.

Since these are new platforms, we can go ahead with always using FlexiBLAS since that's Fedora's preferred way of BLAS switching. So in the `CONFIGURE_OPTIONS`, we should add these same `--with-blas` and `--with-lapack` flags: https://github.com/rstudio/r-builds/blob/03dbd456d48d2e58f22cb5c8afd8b8e092709f50/builder/build.sh#L148-L154

And then in the packaging script, there are some OpenBLAS handling commands that can now be removed.